### PR TITLE
allow `@` expressions

### PR DESCRIPTION
**Pull request by rawwerks**
this commit updates transformers.py to allow `@` expressions. would close https://github.com/zopefoundation/RestrictedPython/issues/268.

### CODE REVIEW

**Comment by icemac** [comment]
Thank you for your pull request.
We are currently asserting that `@` is not supported.
Could you come up with a test asserting that the @-operator works?

**Comment by rawwerks** [comment]
this is a bit above my pay-grade, so please accept my apologies if i am totally misunderstanding how the tests work. 

i'm trying to follow the examples of `test_functiondef.py`...what about something like this?

```
def test_RestrictingNodeTransformer__visit_MatMult__1():
    """It allows the `@` operator for matrix multiplication."""
    source_code = """
class MatMulSupport:
    def __matmul__(self, other):
        return "MatMul Operation Allowed"

def test_matmul():
    a = MatMulSupport()
    b = MatMulSupport()
    return a @ b

result = test_matmul()
"""
    result = compile_restricted_exec(source_code)
    assert result.errors == (funtionandef_err_msg)

```

(i'm so unsure of myself that i'm hoping to get feedback before updating the PR)

**Comment by rawwerks** [comment]
ok wait, this is probably the better place to do it: https://github.com/zopefoundation/RestrictedPython/pull/269/commits/959ac23ee015ba2c4aad0f3e98d1cc502689e350

(maybe not the right way, but i think it should work)

**Comment by icemac** [comment]
I think I understood the idea of the tests and I think I can get it working but first we need some general paperwork:

Thank you for your contribution.

According to the contributing policies of the zopefoundation organization you need to sign a contributor agreement before any non-trivial change can be merged. For details please consult the [Contributing guidelines for zopefoundation projects](https://github.com/zopefoundation/.github/blob/master/CONTRIBUTING.md).

**Comment by rawwerks** [comment]
i'm reviewing the document. 

in the meantime, hopefully we can do this:

this pull request (https://github.com/zopefoundation/RestrictedPython/pull/269) by Raymond Weitekamp (rawwerks) is marked with [CC0 1.0 Universal](http://creativecommons.org/publicdomain/zero/1.0?ref=chooser-v1).

File: tests/transformer/operators/test_arithmetic_operators.py [hunk 1]
```diff
@@ -33,8 +33,16 @@ def test_FloorDiv():
...
+class Vector:
+    def __init__(self, values):
+        self.values = values
+
+    def __matmul__(self, other):
+        return sum(x * y for x, y in zip(self.values, other.values))
+
+result = Vector((8, 3, 5)) @ Vector((2, 7, 1))
+"""
+    # Assuming restricted_eval can execute the source_code and return the value of 'result'
```
Comment (loechel) [inline tests/transformer/operators/test_arithmetic_operators.py:46]: As it is a test in RestrictedPython itself, this comment does not make sense. Whether the test passes and the result is 42 or not, there should not be any assumption.

File: tests/transformer/operators/test_arithmetic_operators.py [hunk 2]
```diff
@@ -33,8 +33,16 @@ def test_FloorDiv():
 
 
 def test_MatMult():
-    result = compile_restricted_eval('(8, 3, 5) @ (2, 7, 1)')
-    assert result.errors == (
-        'Line None: MatMult statements are not allowed.',
-    )
-    assert result.code is None
+    source_code = """
+class Vector:
```
Comment (loechel) [inline tests/transformer/operators/test_arithmetic_operators.py:37]: The test-runner requires it to be:

```python
    source_code = """\
class Vector:
```

**Review by loechel — CHANGES_REQUESTED** [review]
Please fix the linting errors and get the test green.

`transformer.py` Line 55-64 must be ammended and ``__matmul__`` be added, or the tests won't pass.

I would suggest to change the test to the more common used structure, where the source_code definition is definied before the test-case.

Thanks for the Pull request.

**Comment by icemac** [comment]
@rawwerks Thank you for putting your changes into public domain. I'll come up with a new PR containing your changes.
@loechel Please the the changes in https://github.com/zopefoundation/.github/blob/master/CONTRIBUTING.md regarding CC0 contributions.

**Comment by icemac** [comment]
I created #270 as followup PR.

**Comment by loechel** [comment]
@icemac how should I please or approve https://github.com/zopefoundation/.github/pull/8 if it is already merged? I am fine with it, and it makes sense for simple contributions, where people don't want to sign the contributor agreement.

**Comment by icemac** [comment]
@loechel Sorry, I did not check what I wrote, I meant: "Please see the changes in …" in case you were not aware of them.

**Pull request closed without merging**